openwisp_controller/pki/api/serializers.py

```diff
@@ -0,0 +1,184 @@
+from django.utils.translation import ugettext_lazy as _
+from django_x509.base.models import (
+    default_ca_validity_end,
+    default_cert_validity_end,
+    default_validity_start,
+)
+from rest_framework import serializers
+from swapper import load_model
+
+from openwisp_users.api.mixins import FilterSerializerByOrgManaged
+from openwisp_utils.api.serializers import ValidatedModelSerializer
+
+Ca = load_model('django_x509', 'Ca')
+Cert = load_model('django_x509', 'Cert')
+
+
+class BaseSerializer(FilterSerializerByOrgManaged, ValidatedModelSerializer):
+    pass
+
+
+class CaListSerializer(BaseSerializer):
+    key_length = serializers.ChoiceField(
+        allow_blank=False,
+        choices=(('512', '512'), ('1024', '1024'), ('2048', '2048'), ('4096', '4096')),
+        help_text=_('bits'),
+        initial='2048',
+        required=True,
+    )
+    digest = serializers.ChoiceField(
+        allow_blank=False,
+        choices=(
+            ('sha1', 'SHA1'),
+            ('sha224', 'SHA224'),
+            ('sha256', 'SHA256'),
+            ('sha384', 'SHA384'),
+            ('sha512', 'SHA512'),
+        ),
+        help_text=_('bits'),
+        initial='sha256',
+        label='Digest algorithm',
+        required=True,
+    )
```

Comment:
why do we have to redefine this and the other choice fields? Is it really necessary?

Reply:
Hi I redefined it to exclude the unwanted choice field i.e., 'blank' and 'none'. But this will get resolved with this. :- https://github.com/openwisp/openwisp-controller/issues/444 So simplefied the code.